### DISCUSSION TITLE
Generic help with a state trigger

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#397] [Q&A]
This is not a specific question about `pyscript` but rather its use and a request for fresh eyes on a very simple piece of code.

I wanted to write automation that pulls up window covers when someone leaves. It is part of several other automation that works great.  (explanation of the code below the block)

```python
try:
    pyscript.martin_one_time_volets_up
except NameError:
    log.info("setting pyscript.martin_one_time_volets_up to False as it is not defined")
    pyscript.martin_one_time_volets_up = False
state.persist('pyscript.martin_one_time_volets_up')

@state_trigger("person.martin == 'not_home'")
def volets_up_morning():
    log.info(f"martin est parti, {pyscript.martin_one_time_volets_up}, {person.martin}")
    if not pyscript.martin_one_time_volets_up:
        log.info("open martin volets because he left")
        cover.open_cover(entity_id="cover.martin_volets")
        pyscript.martin_one_time_volets_up = True

@time_trigger('once(04:00:00)')
def reset_volets_flag():
    pyscript.martin_one_time_volets_up = False
```

The idea is that when `person.martin` switches to `not_home` some covers get pulled up (via `cover.open_cover(entity_id="cover.martin_volets")`).  Since I want to do it only once, the first time `martin` leaves, I use a flag (`pyscript.martin_one_time_volets_up`) that gets reset every night at 04:00. If the covers are pulled up automatically once, they will not do so another time in one day (I will improve that later).

The problem: I never get into the `if` condition. When forcefully switching `person.martin` to `not_home`, I get the following in the logs:

    domotique-hass-1  | 2022-10-18 11:11:11.050 INFO (MainThread) [custom_components.pyscript.scripts.automation.martin.volets_up_morning] martin est parti, False, not_home

This means that
- the function was triggered (good!)
- `pyscript.martin_one_time_volets_up` is `False` (good! → this is the first time today the script is triggered)
- the state of  `person.martin`  is `not_home` (good! → just a double check but the trigger already tells us so)

**And yet `if not pyscript.martin_one_time_volets_up:` is not stepped into (no logs, and no covers up either)**

This must be something horribly obvious, I apologize in advance if so, but I am looking at this simple code and am stuck.

**Comment by stefanuytterhoeven** [comment]
At first sight, it should work.
What happens when you do if my script.... == False:
